doc/specs/#607 - Commandline Arguments for the Windows Terminal.md

```diff
@@ -0,0 +1,534 @@
+same as `wt new-tab cmd.exe`.
+
+<!--
+### Aside: What should the default command be?
+
+These are notes from my draft intentionally left here to help understand the
+conclusion that new-tab should be the default command.
+
+Should the default command be `new-window` or `new-tab`?
+
+`new-window` makes sense to take params like `--initialPosition`,
+`--initialRows`/`--initialCols`, and _implies_ `new-tab`. However, chained
+commands that want to open in the same window _need_ to specify `new-tab`,
+otherwise they'll all appear in new windows.
+
+If it's `new-tab`, then how do `--initialRows` (etc) work? `new-tab` generally
+_doesn't_ accept those parameters, because it's going to be inheriting the
+parent's window size. Do we just ignore them for subsequent invocations? I
+suppose that makes sense, once the first tab has set those, then the other tabs
+can't really change them.
+
+When dealing with a file full of startup commands, we'll assume all of them are
+intended for the given window. So the first `new-tab` in the file will create
+the window, and all subsequent `new-tab` commands will create tabs in that same
+window.
+ -->
+
+TODO: What should we do with an entirely empty command? My gut says _ignore it_.
+For example, `wt ; ; ; ` should not just open 4 tabs. But then what about `wt`
+by itself? We want that to open a new tab with the default profile. So maybe we
+should allow them? Or allow them on the first command only?
+
+
+### Options
+
+#### `--help,-h,-?`
+Runs the `help` command.
+
+#### `--version,-v`
+Runs the `version` command.
+
+#### `--session,-s session-id`
+Run these commands in the given Windows Terminal session. Enables opening new
+tabs in already running Windows Terminal windows. This feature is dependent upon
+other planned work landing, so is only provided as an example, of what it might
+look like. See [Future Considerations](#Future-Considerations) for more details.
+
+#### `--file,-f congifuration-file`
+Run these commands in the given Windows Terminal session. Enables opening new
+tabs in already running Windows Terminal windows. See [Future
+Considerations](#Future-Considerations) for more details.
+
+### Commands
+
+#### `help`
+
+`help`
+
+Display the help message
+
+#### `version`
+
+`version`
+
+Display version info for the Windows Terminal
+
+#### `open-settings`
+
+`open-settings [--defaults,-d]`
+
+Open the settings file.
+
+**Parameters**:
+* `--defaults,-d`: Open the `defaults.json` file instead of the `profiles.json`
+  file.
+
+#### `list-profiles`
+
+`list-profiles [--all,-A] [--showGuids,-g]`
+
+Displays a list of each of the available profiles. Each profile displays it's
+name, seperated by newlines.
+
+**Parameters**:
+* `--all,-A`: Show all profiles, including profiles marked `"hidden": true`.
+* `--showGuids,-g`:  In addition to showing names, also list each profile's
+  guid. These GUIDs should probably be listed _first_ on each line, to make
+  parsing output easier.

```

Comment:
I'm all for this, but is there a value to getting the guids?

Reply:
Maybe? I guess I really just threw this out there as a proposal. GUIDs are going to be more stable than names, at least as far as scripting is concerned. But you're right that this probably doesn't add real value.

Reply:
I thought the same thing. Not sure if it adds much value.